monkey/agent_plugins/exploiters/powershell/src/plugin.py

```diff
@@ -0,0 +1,142 @@
+import logging
+from functools import partial
+from pprint import pformat
+from typing import Any, Dict, Sequence
+
+# common imports
+from common.event_queue import IAgentEventPublisher
+from common.types import AgentID, Event, NetworkPort, PortStatus
+from common.utils.code_utils import del_key
+from common.utils.environment import is_windows_os
+
+# dependencies to get rid of or internalize
+from infection_monkey.exploit import IAgentBinaryRepository, IAgentOTPProvider
+from infection_monkey.exploit.tools import BruteForceCredentialsProvider, BruteForceExploiter
+from infection_monkey.exploit.tools.helpers import get_agent_dst_path
+from infection_monkey.i_puppet import ExploiterResultData, TargetHost
+from infection_monkey.propagation_credentials_repository import IPropagationCredentialsRepository
+
+from .credentials_generator import generate_powershell_credentials
+from .powershell_command_builder import build_powershell_command
+from .powershell_options import PowerShellOptions
+
+logger = logging.getLogger(__name__)
+
+
+POWERSHELL_NO_SSL_PORT = NetworkPort(5985)
+POWERSHELL_SSL_PORT = NetworkPort(5986)
+
+
+def should_attempt_exploit(host: TargetHost) -> bool:
+    # attempt exploit if any PowerShell remoting port is open or if its status is unknown
+    return (
+        (POWERSHELL_NO_SSL_PORT not in host.ports_status.tcp_ports)
+        or _http_powershell_port_is_open(host)
+    ) or (
+        (POWERSHELL_SSL_PORT not in host.ports_status.tcp_ports)
+        or _https_powershell_port_is_open(host)
+    )
+
+
+def _http_powershell_port_is_open(host: TargetHost) -> bool:
+    return (
+        POWERSHELL_NO_SSL_PORT in host.ports_status.tcp_ports
+        and host.ports_status.tcp_ports[POWERSHELL_NO_SSL_PORT].status == PortStatus.OPEN
+    )
+
+
+def _https_powershell_port_is_open(host: TargetHost) -> bool:
+    return (
+        POWERSHELL_SSL_PORT in host.ports_status.tcp_ports
+        and host.ports_status.tcp_ports[POWERSHELL_SSL_PORT].status == PortStatus.OPEN
+    )
```

Comment:
I believe this is functionally equivalent:
```suggestion
    from infection_monkey.exploit.tools.utils import all_tcp_ports_are_closed  # Put import at the top
    return all_tcp_ports_are_closed(host, [POWERSHELL_NO_SSL_PORT, POWERSHELL_SSL_PORT])
```